simple_use_case/gwl/workflow.w

```diff
@@ -0,0 +1,122 @@
+define domain-size 1
+
+
+process make-gmsh-mesh (with geofile domainsize)
+  synopsis "Generate the computational mesh with gmsh"
+  packages "gmsh"
+  inputs
+    . geofile: (file geofile)
+  outputs
+    . mesh: (file "mesh.msh")
+  # { gmsh -setnumber domainsize {{domainsize}} -2 {{inputs}} -o {{outputs}} }
+
+
+process convert-msh-to-xdmf (with inputmesh)
+  synopsis "Convert the produced gmsh mesh to xdmf"
+  packages "python" "python-meshio"
+  inputs
+    . inputmesh: (file inputmesh)
+  outputs
+    . xdmf: (file (basename inputmesh ".msh") "_converted.xdmf")
+    . h5: (file (basename inputmesh ".msh") "_converted.h5")
+  # { meshio convert {{inputs}} {{outputs:xdmf}} }
+
+
+process run-dolfin (with xdmfmeshfile h5meshfile)
+  synopsis "Run the poisson solver in dolfin"
+  packages "python" "fenics" "pkg-config" "python-pkgconfig" "openmpi"
+    . "openssh" "gcc-toolchain"
+  inputs
+    . script: (file "source/poisson.py")
+    . xdmf: (file xdmfmeshfile)
+    . h5: (file h5meshfile)
+  outputs
+    . pvd: (file "result.pvd")
+    . vtu: (file "result000000.vtu")
+    . num-dof: (file "num-dof")
+  # {
+    python3 {{inputs:script}} --mesh {{inputs:xdmf}} --degree 2 \
+            --output {{outputs:pvd}} --num-dofs {{outputs:num-dof}}
+  }
+
+
+process make-paraview-plot (with pvd vtu)
+  synopsis "Create plot-over-line data with paraview's pvbatch"
+  packages "paraview"
+  inputs
+    . script: (file "source/postprocessing.py")
+    . pvd: (file pvd)
+    . vtu: (file vtu)
+  outputs
+    . csv: (file "plotoverline.csv")
+  # { pvbatch {{inputs:script}} {{inputs:pvd}} {{outputs}} }
+
+
+process prepare-paper-macros (with domain_size num_dofs plot_data_file)
+  synopsis "Update the macros of the paper with values from this run"
+  packages "python"
+  inputs
+    . substitution_script: (file "source/prepare_paper_macros.py")
+    . macros_template: (file "source/macros.tex.template")
+    . num_dofs: (file num_dofs)
+    . plot_data_file: (file plot_data_file)
+  outputs
+    . macros_file: (file "macros.tex")
+  # {
+    python3 {{inputs:substitution_script}} \
+            --domain-size {{domain_size}} \
+            --macro-template-file {{inputs:macros_template}} \
+            --num-dofs $(<{{num_dofs}}) \
+            --plot-data-path {{inputs:plot_data_file}} \
+            --output-macro-file {{outputs:macros_file}}
+  }
+
+process compile-paper (with macros csvfile)
+  synopsis "Create the paper as pdf, given the produced csv file"
+  packages "tectonic" "coreutils"
+  inputs
+    . macros: (file macros)
+    . csvfile: (file csvfile)
+    . texfile: (file "source/paper.tex")
+  outputs
+    . pdf: (file "paper.pdf")
+  # { cp {{inputs:texfile}} paper.tex; tectonic paper.tex }
+
+
+define make-mesh
+  make-gmsh-mesh "source/unit_square.geo" domain-size
+
+
+define convert-mesh
+  convert-msh-to-xdmf
+    pick mesh: (process-outputs make-mesh with-tags:)
```

Comment:
could you explain how information is passed from make-mesh to convert-mesh? Same with the subsequent tasks.

Reply:
Could you add that info as comment in the file, it would really help people understanding the syntax

Reply:
I added some explanatory comments to the workflow.